Backend/Tournament_User/TournamentUserMVSC/controllers/tournamentUserController.js

```diff
@@ -0,0 +1,366 @@
+        if (!tournamentExists) {
+            const error = new Error('Invalid tournamentID: Tournament not found.');
+            error.statusCode = 404;
+            return next(error);
+        }
+
+
+        const GetPlayersInTournament = await TournamentService.getPlayersInTournament(tournamentId);
+        if (GetPlayersInTournament.length === 0) {
+            return res.status(200).json({
+                message: 'No players have signed up for this tournament yet.',
+                success: true,
+                content: []  // Return an empty array
+            });
+        }
+
+        // Map over the result and extract UUID values into an array
+        const uuidArray = GetPlayersInTournament.map(player => player.UUID);
+
+        res.status(200).json({
+            message: 'Players in the tournament:',
+            success: true,
+            content : uuidArray
+        });
+    } catch (err) {
+        console.error('Error retrieving players name:', err);
+        next(err);  // Pass the error to the middleware
+    }
+};
+
+/**
+* Controller to get all tournaments (upcoming, completed, in progress) for a specific player.
+* @param {Object} req - Express request object containing playerUUID in params
+* @param {Object} res - Express response object
+* @param {Function} next - Express next middleware function
+*/
+exports.getPlayerTournaments = async (req, res, next) => {
+    try {
+        const { playerUUID } = req.params;
+        
+        // Validate playerUUID
+        if (!playerUUID) {
+            const error = new Error('Player UUID is required');
+            error.statusCode = 400;
+            return next(error);
+        }
+ 
+        const tournaments = await TournamentUserService.GetPlayerTournamentsByStatus(playerUUID);
+        
+        if (!tournaments || tournaments.length === 0) {
+            const error = new Error('No tournaments found for this player');
+            error.statusCode = 404;
+            return next(error);
+        }
+ 
+        res.status(200).json({
+            "success": true,
+            "status": 200,
+            "message": 'Player tournaments retrieved successfully',
+            "content": tournaments
+        });
+        
+    } catch (err) {
+        console.error('Error retrieving player tournaments:', err);
+        next(err);  // Pass the error to the middleware
+    }
+ };
+
+
+ /**
+* Controller to get all completed tournaments.
+* @param {Object} res - Express response object
+* @param {Function} next - Express next middleware function
+*/
+
+ exports.getCompletedTournaments = async (req, res, next) => {
+    try {
+        const completedTournaments = await TournamentUserService.getCompletedTournaments();
+        if (!completedTournaments || completedTournaments.length === 0) {
+            const error = new Error('No completed tournaments');
+            error.statusCode = 404;
+            return next(error);
```

Comment:
error is handled in the try block instead of throwing an error and handling it in the catch block, am requesting a fix to throw the error n handle it in catch instead

Reply:
oh wait i nvr see this , athirah just added recently